### PR TITLE
Streamlining setters to avoid network call when identifiers are empty

### PR DESCRIPTION
**Pull request by ajaysubra**
# Description

Not making network request when email, phone number and external id is empty or the same as what is in the SDK state.

# Check List

- [ ] Are you changing anything with the public API?
- [ ] Are your changes backwards compatible with previous SDK Versions?
- [ ] Have you tested this change on real device?
- [ ] Have you added unit test coverage for your changes?
- [ ] Have you verified that your changes are compatible with all Android versions the SDK currently supports?

## Related Issues/Tickets

CHNL-7046

### CODE REVIEW

File: sdk/analytics/src/main/java/com/klaviyo/analytics/Klaviyo.kt [hunk 1]
```diff
@@ -97,10 +117,18 @@ object Klaviyo {
      * This should be called whenever the active user in your app changes
      * (e.g. after a fresh login)
      *
+     * note that if the email is empty then the SDK will ignore the email.
+     * use `resetProfile` to clear the email from the profile
+     *
      * @param email Email address for active user
      * @return Returns [Klaviyo] for call chaining
      */
-    fun setEmail(email: String): Klaviyo = this.setProfileAttribute(ProfileKey.EMAIL, email)
```
Comment (evan-masseau) [inline sdk/analytics/src/main/java/com/klaviyo/analytics/Klaviyo.kt:103]: you'll have to wrap all these in `safeApply` like `setProfileAttribute` was.

File: sdk/analytics/src/main/java/com/klaviyo/analytics/Klaviyo.kt [hunk 2]
```diff
@@ -82,9 +84,27 @@ object Klaviyo {
             resetProfile()
         }
 
-        UserInfo.externalId = profile.externalId ?: ""
-        UserInfo.email = profile.email ?: ""
-        UserInfo.phoneNumber = profile.phoneNumber ?: ""
+        if (profile.email != null &&
```
Comment (evan-masseau) [inline sdk/analytics/src/main/java/com/klaviyo/analytics/Klaviyo.kt:87]: I think there might be a cleaner place for this to live, in `ProfileOperationQueue`

File: sdk/analytics/src/test/java/com/klaviyo/analytics/KlaviyoTest.kt [hunk 3]
```diff
@@ -188,6 +188,36 @@ internal class KlaviyoTest : BaseTest() {
...
+        verifyProfileDebounced()
+    }
+
+    @Test
+    fun `SetProfile's fluent setters ignores empty strings and doesn't enqueue profile request`() {
+        Klaviyo.setProfile(Profile().setEmail("").setPhoneNumber("").setExternalId(""))
+
+        assertEquals(Klaviyo.getEmail(), null)
+        assertEquals(Klaviyo.getPhoneNumber(), null)
+        assertEquals(Klaviyo.getExternalId(), null)
```
Comment (evan-masseau) [inline sdk/analytics/src/test/java/com/klaviyo/analytics/KlaviyoTest.kt:218]: add assertion?
```
        verify(exactly = 0) { apiClientMock.enqueueProfile(any()) }
```

File: sdk/analytics/src/main/java/com/klaviyo/analytics/model/Profile.kt [hunk 4]
```diff
@@ -14,21 +14,27 @@ class Profile(properties: Map<ProfileKey, Serializable>?) :
     var externalId: String?
         get() = (this[ProfileKey.EXTERNAL_ID])?.toString()
         set(value) {
-            this[ProfileKey.EXTERNAL_ID] = value
```
Comment (evan-masseau) [inline sdk/analytics/src/main/java/com/klaviyo/analytics/model/Profile.kt:17]: i don't think you need to change `Profile` at all. This is just a data class, if for some reason you need to set email back to null you can. This doesn't affect state.

File: sdk/analytics/src/test/java/com/klaviyo/analytics/KlaviyoTest.kt [hunk 5]
```diff
@@ -235,6 +305,26 @@ internal class KlaviyoTest : BaseTest() {
...
+        val initialPhone = "+12223334444"
+        UserInfo.phoneNumber = initialPhone
+
+        Klaviyo.setPhoneNumber("")
+
+        verify(exactly = 0) { apiClientMock.enqueueProfile(any()) }
+    }
+
+    @Test
+    fun `setPhoneNumber does not update UserInfo when called with the same phone number`() {
```
Comment (evan-masseau) [inline sdk/analytics/src/test/java/com/klaviyo/analytics/KlaviyoTest.kt:319]:
```suggestion
    fun `setPhoneNumber does not enqueue API Request when called with the same phone number`() {
```

File: sdk/analytics/src/test/java/com/klaviyo/analytics/KlaviyoTest.kt [hunk 6]
```diff
@@ -227,6 +277,26 @@ internal class KlaviyoTest : BaseTest() {
...
+        UserInfo.email = EMAIL
+
+        Klaviyo.setEmail("")
+        assertEquals(EMAIL, UserInfo.email)
+
+        verify(exactly = 0) { apiClientMock.enqueueProfile(any()) }
+    }
+
+    @Test
+    fun `setEmail does not update UserInfo when called with the same email`() {
```
Comment (evan-masseau) [inline sdk/analytics/src/test/java/com/klaviyo/analytics/KlaviyoTest.kt:291]:
```suggestion
    fun `setEmail does not enqueue API request when called with the same email`() {
```

File: sdk/analytics/src/test/java/com/klaviyo/analytics/KlaviyoTest.kt [hunk 7]
```diff
@@ -227,6 +277,26 @@ internal class KlaviyoTest : BaseTest() {
         verifyProfileDebounced()
     }
 
+    @Test
+    fun `Sets empty email into info doesn't enqueue profile request`() {
```
Comment (evan-masseau) [inline sdk/analytics/src/test/java/com/klaviyo/analytics/KlaviyoTest.kt:281]:
```suggestion
    fun `setEmail with empty string does not alter state`() {
```

File: sdk/analytics/src/main/java/com/klaviyo/analytics/Klaviyo.kt (hunk 2 above)
Comment (evan-masseau) [reply sdk/analytics/src/main/java/com/klaviyo/analytics/Klaviyo.kt:87]: Then we'd only have to check idenfitiers in one place, no code changes need in `Klaviyo.kt`

**Comment by ajaysubra** [comment]
Closing this as android doesn't send empty identifiers and @evan-masseau has another PR coming soon to dedup profile identifiers.

**Pull request closed without merging**